File: leetcode/String/Roman to Integer(13)/Solution.java
class Solution {
    public int romanToInt(String s) {
        int result = 0;
        int prevSymbol = -1;
        for (int i = s.length()-1; i >= 0; i--) {
            int symbol = symbolToInt(s.charAt(i));
            if(prevSymbol>symbol) {
                result-=symbol;
                prevSymbol = -1;
                continue;
            }else {
                result+=symbol;
            }
            prevSymbol = symbol;
        }
        return result;
    }

    public static int symbolToInt(char c) {
        switch (c) {
            case 'I' -> {
                return 1;
            }
            case 'V' -> {
                return 5;
            }
            case 'X' -> {
                return 10;
            }
            case 'L' -> {
                return 50;
            }
            case 'C' -> {
                return 100;
            }
            case 'D' -> {
                return 500;
            }
            case 'M' -> {
                return 1000;
            }
        }
        return -1;
    }
}

File: leetcode/String/Roman to Integer(13)/Solution.kt
class Solution {
    fun romanToInt(s: String): Int {
        var result = 0
        var prevSymbol = -1
        for (i in s.length -1 downTo 0) {
            val symbol = symbolToINt(s[i])
            if (prevSymbol>symbol) {
                result -= symbol
                prevSymbol = -1
            }else {
                result += symbol
                prevSymbol = symbol
            }
        }

        return result
    }

     fun symbolToINt(c: Char): Int {
        when(c) {
            'I' -> return 1
            'V' -> return 5
            'X' -> return 10
            'L' -> return 50
            'C' -> return 100
            'D' -> return 500
            'M' -> return 1000
        }
        return 0
    }
}
